Website/Providers/DataProviders/SqlDataProvider/09.04.00.01.SqlDataProvider

```diff
@@ -0,0 +1,168 @@
+	DROP PROCEDURE {databaseOwner}[{objectQualifier}UserAgreedToTerms]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}UserAgreedToTerms]
+	@PortalId	INT,
+	@UserId		INT
+AS
+	UPDATE {databaseOwner}{objectQualifier}UserPortals
+	SET HasAgreedToTerms = 1, HasAgreedToTermsOn = GETDATE()
+	WHERE PortalId = @PortalId AND UserId = @UserId
+GO
+
+IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}ResetTermsAgreement]') AND type in (N'P', N'PC'))
+	DROP PROCEDURE {databaseOwner}[{objectQualifier}ResetTermsAgreement]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}ResetTermsAgreement]
+	@PortalId	INT
+AS
+	UPDATE {databaseOwner}{objectQualifier}UserPortals
+	SET HasAgreedToTerms = 0
+	WHERE PortalId = @PortalId
+GO
+
+/*      Add logging for consent             */
+/********************************************/
+
+INSERT INTO {databaseOwner}[{objectQualifier}EventLogTypes] ([LogTypeKey], [LogTypeFriendlyName], [LogTypeDescription], [LogTypeOwner], [LogTypeCSSClass]) 
+VALUES (N'DATA_CONSENT_SUCCESS', N'User Consented', N'', N'DotNetNuke.Logging.EventLogType', N'OperationSuccess')
+GO
+
+INSERT INTO {databaseOwner}[{objectQualifier}EventLogTypes] ([LogTypeKey], [LogTypeFriendlyName], [LogTypeDescription], [LogTypeOwner], [LogTypeCSSClass]) 
+VALUES (N'DATA_CONSENT_FAILURE', N'User Failed to Consent', N'', N'DotNetNuke.Logging.EventLogType', N'AdminAlert')
+GO
+
+IF NOT EXISTS (SELECT * FROM {databaseOwner}{objectQualifier}EventLogConfig WHERE LogTypeKey = 'DATA_CONSENT_SUCCESS')
+BEGIN  
+INSERT INTO {databaseOwner}[{objectQualifier}EventLogConfig] ( [LogTypeKey], [LogTypePortalID], [LoggingIsActive], [KeepMostRecent], [EmailNotificationIsActive], [NotificationThreshold], [NotificationThresholdTime], [NotificationThresholdTimeType], [MailFromAddress], [MailToAddress]) 
+VALUES ( N'DATA_CONSENT_SUCCESS', NULL, 1, 10, 0, 1, 1, 1, N'', N'')
+END
+GO
+
+IF NOT EXISTS (SELECT * FROM {databaseOwner}{objectQualifier}EventLogConfig WHERE LogTypeKey = 'DATA_CONSENT_FAILURE')
+BEGIN  
+INSERT INTO {databaseOwner}[{objectQualifier}EventLogConfig] ( [LogTypeKey], [LogTypePortalID], [LoggingIsActive], [KeepMostRecent], [EmailNotificationIsActive], [NotificationThreshold], [NotificationThresholdTime], [NotificationThresholdTimeType], [MailFromAddress], [MailToAddress]) 
+VALUES ( N'DATA_CONSENT_FAILURE', NULL, 1, 10, 0, 1, 1, 1, N'', N'')
+END
+GO
+
+/*    Used for anonymizing an account       */
+/********************************************/
+
+IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}DeleteUserProfile]') AND type in (N'P', N'PC'))
+	DROP PROCEDURE {databaseOwner}[{objectQualifier}DeleteUserProfile]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}DeleteUserProfile]
+	@PortalId	INT,
+	@UserId INT
+AS
+    IF EXISTS (SELECT * FROM {databaseOwner}{objectQualifier}UserProfile up 
```

Comment:
IMO this distinction is not correct, it doesn't cover superuser profiles properly.
You should distinguish upon @PortalID parameter, whether to delete portal specific properties (positive value) or all properties (PortalID is Null or -1). It depends on Userportals records, whether to call with portalID (other userportals record with other portalID exists) or without (otherwise).

Reply:
I can't tell what this is about.